### PR TITLE
RUN-2343: testdesk webhook migration

### PR DESCRIPTION
**Pull request by alexander-variacode**
## ⛔️ Ticket:
https://pagerduty.atlassian.net/browse/RUN-2343

## ✅ Description:
QA Framework webhook test migration to testdesk webhook test

### CODE REVIEW

File: functional-test/src/test/groovy/org/rundeck/util/gui/pages/project/ProjectExportPage.groovy [hunk 1]
```diff
@@ -18,6 +18,20 @@ class ProjectExportPage extends BasePage {
...
     By stripNameBy = By.xpath("//label[contains(@for,'trip')]")
     By checkBoxBy = By.xpath("//*[@type='checkbox' and not(@id='preserveuuid')]")
+    By allCheckboxBy = By.id('exportAll')
+    By exportJobsCheckboxBy = By.id('exportJobs')
+    By exportExecutionsCheckboxBy = By.id('exportExecutions')
+    By exportConfigsCheckboxBy = By.id('exportConfigs')
+    By exportReadmesCheckboxBy = By.id('exportReadmes')
+    By exportAclsCheckboxBy = By.id('exportAcls')
+    By exportScmCheckboxBy = By.id('exportScm')
+    By tourManagerCheckboxBy = By.id('exportComponents.tours-manager')
```
Comment (ronaveva) [inline functional-test/src/test/groovy/org/rundeck/util/gui/pages/project/ProjectExportPage.groovy:28]: these locators apply only to PRO, it should not be in OSS code

File: functional-test/src/test/resources/projects-import/webhooks-test/rundeck-importProjectWithWebhook/webhooks.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+webhooks:
```
Comment (ronaveva) [inline functional-test/src/test/resources/projects-import/webhooks-test/rundeck-importProjectWithWebhook/webhooks.yaml:1]: if this is only used in PRO, it should go in PRO repo

**Review by ronaveva — CHANGES_REQUESTED** [review]
please check de comments

File: functional-test/src/test/groovy/org/rundeck/util/gui/pages/project/ProjectExportPage.groovy (hunk 1 above)
Comment (alexander-variacode) [reply functional-test/src/test/groovy/org/rundeck/util/gui/pages/project/ProjectExportPage.groovy]: Done...

File: functional-test/src/test/resources/projects-import/webhooks-test/rundeck-importProjectWithWebhook/webhooks.yaml (hunk 2 above)
Comment (alexander-variacode) [reply functional-test/src/test/resources/projects-import/webhooks-test/rundeck-importProjectWithWebhook/webhooks.yaml:1]: it stays in oss

File: functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/webhook/WebhooksSpec.groovy [hunk 3]
```diff
@@ -36,4 +41,181 @@ class WebhooksSpec extends SeleniumBase {
...
+        setupProject(projectName)
+        WebhooksPage webhookPage = page WebhooksPage
+        LoginPage loginPage = go LoginPage
+        HomePage homePage = page HomePage
+
+        when:
+        loginPage.login(TEST_USER, TEST_PASS)
+        homePage.validatePage()
+        webhookPage.loadPageForProject(projectName)
+        createSimpleWebhook(webhookPage, webhookName)
```
Comment (ronaveva) [inline functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/webhook/WebhooksSpec.groovy:60]: here it should also validate the webhook to exist

File: functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/webhook/WebhooksSpec.groovy [hunk 4]
```diff
@@ -36,4 +41,181 @@ class WebhooksSpec extends SeleniumBase {
...
+        webhookPage.waitForNumberOfElementsToBe(webhookPage.alertInfoBy, 0)
+        removeWebhook(webhookPage, webhookName)
```
Comment (ronaveva) [inline functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/webhook/WebhooksSpec.groovy:62]: here it should validate that the webhook does not exist

**Review by ronaveva — CHANGES_REQUESTED** [review]
Overall looks ok, just minor validations missing (check the comments)

File: functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/webhook/WebhooksSpec.groovy (hunk 3 above)
Comment (alexander-variacode) [reply functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/webhook/WebhooksSpec.groovy:60]: Done...✅

File: functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/webhook/WebhooksSpec.groovy (hunk 4 above)
Comment (alexander-variacode) [reply functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/webhook/WebhooksSpec.groovy:62]: Done...✅

File: functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/webhook/WebhooksSpec.groovy [hunk 5]
```diff
@@ -36,4 +41,183 @@ class WebhooksSpec extends SeleniumBase {
...
+        setupProject(projectName)
+        WebhooksPage webhookPage = page WebhooksPage
+        LoginPage loginPage = go LoginPage
+        HomePage homePage = page HomePage
+
+        when:
+        loginPage.login(TEST_USER, TEST_PASS)
+        homePage.validatePage()
+        webhookPage.loadPageForProject(projectName)
+        createSimpleWebhook(webhookPage, webhookName)
```
Comment (ronaveva) [inline functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/webhook/WebhooksSpec.groovy:60]:
```suggestion
        createSimpleWebhook(webhookPage, webhookName)
        then: "checks for the WH to exist"
        containsName(webhookPage, webhookName)
        when: "remove the wh"
```
this way the validation is more explicit and easy to read, like after deleting the wh

**Review by ronaveva — APPROVED** [review]
LGTM

**Pull request merged**